internal/bminventory/inventory.go

```diff
@@ -1956,6 +1956,14 @@ func (b *bareMetalInventory) updateClusterData(ctx context.Context, cluster *com
 		}
 	}
 
+	if params.ClusterUpdateParams.MachineNetworkCidr != nil && common.IsSingleNodeCluster(cluster) {
```

Comment:
Why not set it in line 1936?

Reply:
cause i will not be there in that case

Reply:
So add "or" (||) as part of the condition

Reply:
OK